internal/session/session.go

```diff
@@ -183,74 +182,94 @@ func (s *Session) Shutdown() {
 	s.shutdown()
 }
 
-// ReceiveFrom receives incoming blocks from the given peer.
-func (s *Session) ReceiveFrom(from peer.ID, ks []cid.Cid, haves []cid.Cid, dontHaves []cid.Cid) {
-	// The SessionManager tells each Session about all keys that it may be
-	// interested in. Here the Session filters the keys to the ones that this
-	// particular Session is interested in.
-	interestedRes := s.sim.FilterSessionInterested(s.id, ks, haves, dontHaves)
-	ks = interestedRes[0]
-	haves = interestedRes[1]
-	dontHaves = interestedRes[2]
-	s.logReceiveFrom(from, ks, haves, dontHaves)
+// GetBlock fetches a single block.
+func (s *Session) GetBlock(reqctx context.Context, k cid.Cid) (blocks.Block, error) {
```

Comment:
The code in GetBlock() was moved from getter.go into the Session. Seems like a more natural place for it to live.